
# HG changeset patch
# User Timo Sirainen <tss@iki.fi>
# Date 1233425646 -7200
# Node ID 498022697a33e7c6a4db2e9e7db763ffc5552872
# Parent f97cd7e3acda5a4b707cde9837c55a8cd48d2d8d
auth: Using "username" or "domain" passdb fields caused problems with cache and blocking passdbs.

--- a/src/auth/auth-request.c	Thu Jan 29 19:25:02 2009 -0500
+++ b/src/auth/auth-request.c	Sat Jan 31 20:14:06 2009 +0200
@@ -1006,7 +1006,7 @@ void auth_request_set_field(struct auth_
 			    const char *name, const char *value,
 			    const char *default_scheme)
 {
-	const char *p;
+	const char *p, *orig_value;
 
 	i_assert(*name != '\0');
 	i_assert(value != NULL);
@@ -1024,6 +1024,7 @@ void auth_request_set_field(struct auth_
 	if (strcmp(name, "user") == 0 ||
 	    strcmp(name, "username") == 0 || strcmp(name, "domain") == 0) {
 		/* update username */
+		orig_value = value;
 		if (strcmp(name, "username") == 0 &&
 		    strchr(value, '@') == NULL &&
 		    (p = strchr(request->user, '@')) != NULL) {
@@ -1054,6 +1055,9 @@ void auth_request_set_field(struct auth_
 				request->user, value);
 			request->user = p_strdup(request->pool, value);
 		}
+		/* restore the original value so it gets saved correctly to
+		   cache. */
+		value = orig_value;
 	} else if (strcmp(name, "nodelay") == 0) {
 		/* don't delay replying to client of the failure */
 		request->no_failure_delay = TRUE;

